src/datachain/query/dispatch.py

```diff
@@ -196,14 +196,17 @@ def run_udf_single(
         generated_cb: Callback = DEFAULT_CALLBACK,
     ) -> None:
         udf: UDFAdapter = loads(self.udf_data)
-
         if ids_only and not self.is_batching:
             input_rows = flatten(input_rows)
 
         def get_inputs() -> Iterable["RowsOutput"]:
             warehouse = self.catalog.warehouse.clone()
             if ids_only:
-                for ids in batched(input_rows, DEFAULT_BATCH_SIZE):
```

Comment:
Please correct me if I'm wrong, but this should not be connected to the issue.

Reply:
It could be helpful to read the rows in chunk from warehouse that can be customized,. But, reverted back for now,